File: anime_coloring.cs
// anime_coloring.cs — C# версия

using System;
using System.Collections.Generic;
using System.IO;

class AnimeColoring
{
    static Dictionary<string, int> COLORS = new Dictionary<string, int>
    {
        {"black", 30}, {"red", 31}, {"green", 32}, {"yellow", 33},
        {"blue", 34}, {"magenta", 35}, {"cyan", 36}, {"white", 37},
        {"gray", 90}, {"bright_red", 91}, {"bright_green", 92},
        {"bright_yellow", 93}, {"bright_blue", 94}, {"bright_magenta", 95},
        {"bright_cyan", 96}, {"bright_white", 97}
    };

    static string TEMPLATE = @"
          {hair}██{hair}██{hair}██{hair}██{hair}██
        {hair}██{skin}      {skin}{hair}██
       {hair}██  {eye}██{hair}  {hair}██  {eye}██{hair}  {hair}██
       {hair}██  {eye}██{hair}  {hair}██  {eye}██{hair}  {hair}██
        {hair}██{skin}      {skin}{hair}██
         {hair}██{hair}██{hair}██{hair}██{hair}██{hair}██{hair}██{hair}
          {hair}██  {hair}██
         {hair}██{clothes}    {clothes}{hair}██
         {hair}██{clothes}    {clothes}{hair}██
          {hair}██{hair}██{hair}██{hair}██{hair}██";

    static string Colorize(string text, string colorName)
    {
        if (!COLORS.TryGetValue(colorName, out int code))
            code = 37;
        return $"\u001B[{code}m{text}\u001B[0m";
    }

    static string GenerateImage(string hair, string eyes, string skin, string clothes)
    {
        string result = TEMPLATE;
        result = result.Replace("{hair}", Colorize("██", hair));
        result = result.Replace("{eyes}", Colorize("██", eyes));
        result = result.Replace("{skin}", Colorize("  ", skin));
        result = result.Replace("{clothes}", Colorize("██", clothes));
        return result;
    }

    static void Main(string[] args)
    {
        string hair = "magenta", eyes = "cyan", skin = "yellow", clothes = "blue";
        string output = null;

        for (int i = 0; i < args.Length; i++)
        {
            if (args[i] == "--hair") hair = args[++i];
            else if (args[i] == "--eyes") eyes = args[++i];
            else if (args[i] == "--skin") skin = args[++i];
            else if (args[i] == "--clothes") clothes = args[++i];
            else if (args[i] == "--output") output = args[++i];
        }

        Console.WriteLine("🎨 Anime Coloring (C#)");
        Console.WriteLine($"Цвета: волосы={hair}, глаза={eyes}, кожа={skin}, одежда={clothes}");
        Console.WriteLine();
        string image = GenerateImage(hair, eyes, skin, clothes);
        Console.WriteLine(image);

        if (output != null)
        {
            File.WriteAllText(output, image + "\n");
            Console.WriteLine($"💾 Сохранено в {output}");
        }
    }
}

File: AnimeColoring.java
// AnimeColoring.java — Java версия

import java.util.HashMap;
import java.util.Map;

public class AnimeColoring {
    private static final Map<String, Integer> COLORS = new HashMap<>();
    static {
        COLORS.put("black", 30); COLORS.put("red", 31);
        COLORS.put("green", 32); COLORS.put("yellow", 33);
        COLORS.put("blue", 34); COLORS.put("magenta", 35);
        COLORS.put("cyan", 36); COLORS.put("white", 37);
        COLORS.put("gray", 90); COLORS.put("bright_red", 91);
        COLORS.put("bright_green", 92); COLORS.put("bright_yellow", 93);
        COLORS.put("bright_blue", 94); COLORS.put("bright_magenta", 95);
        COLORS.put("bright_cyan", 96); COLORS.put("bright_white", 97);
    }

    private static final String TEMPLATE = 
        "\n          {hair}██{hair}██{hair}██{hair}██{hair}██" +
        "\n        {hair}██{skin}      {skin}{hair}██" +
        "\n       {hair}██  {eye}██{hair}  {hair}██  {eye}██{hair}  {hair}██" +
        "\n       {hair}██  {eye}██{hair}  {hair}██  {eye}██{hair}  {hair}██" +
        "\n        {hair}██{skin}      {skin}{hair}██" +
        "\n         {hair}██{hair}██{hair}██{hair}██{hair}██{hair}██{hair}██{hair}" +
        "\n          {hair}██  {hair}██" +
        "\n         {hair}██{clothes}    {clothes}{hair}██" +
        "\n         {hair}██{clothes}    {clothes}{hair}██" +
        "\n          {hair}██{hair}██{hair}██{hair}██{hair}██";

    private static String colorize(String text, String colorName) {
        Integer code = COLORS.get(colorName);
        if (code == null) code = 37;
        return "\u001B[" + code + "m" + text + "\u001B[0m";
    }

    private static String generateImage(String hair, String eyes, String skin, String clothes) {
        String result = TEMPLATE;
        result = result.replace("{hair}", colorize("██", hair));
        result = result.replace("{eyes}", colorize("██", eyes));
        result = result.replace("{skin}", colorize("  ", skin));
        result = result.replace("{clothes}", colorize("██", clothes));
        return result;
    }

    public static void main(String[] args) {
        String hair = "magenta", eyes = "cyan", skin = "yellow", clothes = "blue";
        String output = null;

        for (int i = 0; i < args.length; i++) {
            if (args[i].equals("--hair")) hair = args[++i];
            else if (args[i].equals("--eyes")) eyes = args[++i];
            else if (args[i].equals("--skin")) skin = args[++i];
            else if (args[i].equals("--clothes")) clothes = args[++i];
            else if (args[i].equals("--output")) output = args[++i];
        }

        System.out.println("🎨 Anime Coloring (Java)");
        System.out.printf("Цвета: волосы=%s, глаза=%s, кожа=%s, одежда=%s\n", hair, eyes, skin, clothes);
        System.out.println();
        String image = generateImage(hair, eyes, skin, clothes);
        System.out.println(image);

        if (output != null) {
            try (java.io.FileWriter fw = new java.io.FileWriter(output)) {
                fw.write(image + "\n");
                System.out.println("💾 Сохранено в " + output);
            } catch (Exception e) {
                System.err.println("Ошибка сохранения: " + e.getMessage());
            }
        }
    }
}
